### ISSUE TITLE
Distorted font with non-standard monitor DPI setting

### ISSUE THREAD

**Issue by creaothceann** [floooh/v6502r#27]
Whenever I move the browser window containing visual6502r to a monitor with a DPI setting <> 100% (via "Settings | Display | Scale & layout") the font becomes distorted, making it very unpleasant to read.

OS: Windows 11 Pro 23H2
browsers: Firefox 126.0.1 / Chrome 126.0.6478.115
monitor 1: size=3840x2160, scale=150%
monitor 2: size=2560x1440, scale=100%

monitor 1:
![visual6502r_1](https://github.com/floooh/v6502r/assets/1885014/167dcdde-d9cb-4026-bbce-7a2526dda43f)

monitor 2:
![visual6502r_2](https://github.com/floooh/v6502r/assets/1885014/f7d9e390-0bd1-462d-ac55-c23e6e229d02)

**Comment by floooh** [comment]
Yeah that's a known issue and unrelated to the v6502r. With fractional scaling, one can either have crisp text with scaling artifacts, or blurry text. Does it looks better when starting with this URL?

https://floooh.github.io/visualz80remix/?highdpi

**Comment by creaothceann** [comment]
The first link looks basically the same; the second link doesn't have the problem.

Would it be possible to add a theme and/or checkbox that enables the font and filtering? I didn't see an option for that in the menu.

**Comment by floooh** [comment]
Currently that setting can't be changed after start, but I guess it's a good idea to make it the default.

As a temporary measure, just append a `?highdpi` to the URL, e.g.

- Z80: https://floooh.github.io/visualz80remix/?highdpi
- 6502: https://floooh.github.io/visual6502remix/?highdpi
- 2A03: https://floooh.github.io/visual2a03remix/?highdpi

**Comment by floooh** [comment]
Deleted a previous comment (I thought it's the highdpi setting that looks good, not the different font and filtering).

Just plugging the font from the second demo into the app won't work unfortunately because it needs to be a monospace font, otherwise formatting in the text editor, and the layout in the debugging editor will break. I'll need to check how it looks keeping the default monospace font but enabling high-dpi and linear filtering for the text rendering. It will definitely look blurry but that may be the less bad option for fractional scaling. I won't have time for this anytime soon though, but I'll keep the ticket open as reminder.

**Comment by creaothceann** [comment]
Thanks! While I can just use the second monitor (or [a VM...](https://i.imgur.com/RCB6Xyw.png)) it will probably matter for other users.
